### ISSUE TITLE
Wix error when using multiple directories

### ISSUE THREAD

**Issue by bradh** [oleg-shilo/wixsharp#7]
With the following code:
```C#
using WixSharp;

namespace WixSharpSetup
{
    class Program
    {
        static void Main()
        {
            var project = new ManagedProject()
            {
                Name = "MyProduct",
                Dirs = new[]
                {
                    new Dir("%ProgramFilesFolder%/Foo"),
                    new Dir(@"C:\something\else")
                }
            };
            project.BuildMsi();
        }
    }
}
```

The build ends with:

> 
>  Wix project file has been built: c:\users\admin\documents\visual studio 2015\Projects\ClassLibrary1\WixSharp Setup1\setup.wxs
> 1>  
> 1>  Windows Installer XML Toolset Compiler version 3.10.3.3007
> 1>  Copyright (c) .NET Foundation and contributors. All rights reserved.
> 1>  
> 1>  setup.wxs
> 1>c:\users\admin\documents\visual studio 2015\Projects\ClassLibrary1\WixSharp Setup1\setup.wxs(19): error CNDL0027: The Directory/@Name attribute's value, 'C:\something\else', is not a valid long name because it contains illegal characters.  Legal long names contain no more than 260 characters and must contain at least one non-period character.  Any character except for the follow may be used: \ ? | > < : / * ".
> 1>  Cannot build c:\users\admin\documents\visual studio 2015\Projects\ClassLibrary1\WixSharp Setup1\setup.wxs
> 1>c:\users\admin\documents\visual studio 2015\Projects\ClassLibrary1\packages\WixSharp.1.2.0.0\build\WixSharp.targets(6,5): error MSB3073: The command ""c:\users\admin\documents\visual studio 2015\Projects\ClassLibrary1\WixSharp Setup1\bin\Debug\WixSharpSetup.exe" "/MSBUILD:WixSharp Setup1"" exited with code -1.

The `.wxs` file ends up with:
```XML
    <Directory Id="TARGETDIR" Name="SourceDir">
      <Directory Id="ProgramFilesFolder" Name="ProgramFilesFolder">
        <Directory Id="INSTALLDIR" Name="Foo">

          <Component Id="INSTALLDIR.EmptyDirectory" Guid="73a032e4-80fb-4301-bde9-58247c74d6ad" KeyPath="yes">
            <CreateFolder />
            <RemoveFolder Id="INSTALLDIR" On="uninstall" />
          </Component>

        </Directory>
      </Directory>

      <Directory Id="TARGETDIR.C_.something.else" Name="C:\something\else">

        <Component Id="else.EmptyDirectory" Guid="73a032e4-80fb-4301-bde9-5824ae4ac65b" KeyPath="yes">
          <CreateFolder />
          <RemoveFolder Id="TARGETDIR.C_.something.else" On="uninstall" />
        </Component>

      </Directory>
```

I am currently working around this using a load-time hook (which was actually more elegant anyway), but it looks like a possible bug to allow multiple entries in `.Dirs` but for the `Name` attribute to be invalid.

**Comment by oleg-shilo** [comment]
Thanks, Yes it is a defect. 
Fixed with #a30d2d49c8b21b9d2eb2e242379d86a6fc7383a2 The binaries will be available with the next release. Alternatively you can grab them from https://github.com/oleg-shilo/wixsharp/tree/master/Source/src/WixSharp.Samples

>I am currently working around this using a load-time hook (which was actually more elegant anyway)

Actually it doesn't surprise my. I am glad you find it elegant. 
Many consider direct XML manipulation as something hacky, while... it's truly not. Wix# compiler emits significant amount of WXS this way. The compiler events are just giving the developer the opportunity to shape XML with the additional (non-built-in) routines. That's it. But of course if the routine is useful then it should become the default functionality.

**Comment by oleg-shilo** [comment]
Fixed in release v1.3
